cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1155,47 +1167,19 @@ def configure_apiserver(etcd_connection_string, leader_etcd_version):
         'DefaultTolerationSeconds'
     ]
 
-    admission_control = [
-        'NamespaceLifecycle',
-        'LimitRanger',
-        'ServiceAccount',
-        'PersistentVolumeLabel',
-        'DefaultStorageClass',
-        'DefaultTolerationSeconds',
-        'MutatingAdmissionWebhook',
-        'ValidatingAdmissionWebhook',
-        'ResourceQuota'
-    ]
-
     auth_mode = hookenv.config('authorization-mode')
     if 'Node' in auth_mode:
         admission_control.append('NodeRestriction')
 
     api_opts['authorization-mode'] = auth_mode
 
-    kube_version = get_version('kube-apiserver')
-    if kube_version < (1, 6):
+    if get_version('kube-apiserver') < (1, 6):
         hookenv.log('Removing DefaultTolerationSeconds from admission-control')
-        admission_control_pre_1_9.remove('DefaultTolerationSeconds')
-    if kube_version < (1, 7):
+        admission_control.remove('DefaultTolerationSeconds')
+    if get_version('kube-apiserver') < (1, 7):
         hookenv.log('Removing Initializers from admission-control')
-        admission_control_pre_1_9.remove('Initializers')
-    if kube_version < (1, 9):
-        api_opts['admission-control'] = ','.join(admission_control_pre_1_9)
-    else:
-        api_opts['admission-control'] = ','.join(admission_control)
-
-    if get_version('kube-apiserver') > (1, 6) and \
-       hookenv.config('enable-metrics'):
-        api_opts['requestheader-client-ca-file'] = ca_cert_path
-        api_opts['requestheader-allowed-names'] = 'client'
-        api_opts['requestheader-extra-headers-prefix'] = 'X-Remote-Extra-'
-        api_opts['requestheader-group-headers'] = 'X-Remote-Group'
-        api_opts['requestheader-username-headers'] = 'X-Remote-User'
-        api_opts['proxy-client-cert-file'] = client_cert_path
-        api_opts['proxy-client-key-file'] = client_key_path
-        api_opts['enable-aggregator-routing'] = 'true'
-        api_opts['client-ca-file'] = ca_cert_path
+        admission_control.remove('Initializers')
+    api_opts['admission-control'] = ','.join(admission_control)
```

Comment:
A lot of accidental changes to api_opts and admission_control here, I think

Reply:
Major fail...